ngraph/core/include/ngraph/op/constant.hpp

```diff
@@ -17,6 +16,26 @@
 #include "ngraph/type/element_type_traits.hpp"
 #include "ngraph/util.hpp"
 
+namespace
+{
+    template <ngraph::element::Type_t Type,
+              typename std::enable_if<Type == ngraph::element::Type_t::u4, bool>::type = true>
+    ngraph::fundamental_type_for<Type>
+        value_in_range(const ngraph::fundamental_type_for<Type>& value)
+    {
+        NGRAPH_CHECK(0 <= value && value <= 15, "assigned value out of range u4 values");
+        return value;
+    }
+
+    template <ngraph::element::Type_t Type,
+              typename std::enable_if<Type == ngraph::element::Type_t::i4, bool>::type = true>
+    ngraph::fundamental_type_for<Type>
+        value_in_range(const ngraph::fundamental_type_for<Type>& value)
+    {
+        NGRAPH_CHECK(-8 <= value && value <= 7, "assigned value out of range i4 values");
```

Comment:
I didn't find any tests to cover these checks. Can you add tests to cover out of range cases?

Reply:
OK